File: school/summer2013/openmp/c/43_offload_n_openmp.c
#include <stdio.h>

int main()
{
	long i, sum=0;
#pragma omp parallel
{
	#pragma omp single
	#pragma offload target(mic)
	{
		printf("check\n");
	}
	#pragma omp for reduction(+:sum)
	for(i=1; i<=N; i++)
		sum += i;
}
	printf("sum = %ld\n", sum);
}

File: summer2013/example_openmp/fortran/43_offload_n_openmp.f90
program sum_offload_n_openmp2
    use omp_lib
    implicit none
    integer(8) :: i, total_sum = 0

!$omp parallel
    !$omp single
        !dir$ offload begin target(mic)
            print *, 'check'
        !dir$ end offload
    !$omp end single

    !$omp do reduction (+:total_sum)
    do i=1,N
        total_sum = total_sum + i
    end do
    !$omp end do
!$omp end parallel
end
